### PR TITLE
[11.0][MIG] product

### PR DESCRIPTION
**Pull request by Benniphx**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/product/migrations/11.0.1.2/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            """, (product_rows[0], r[0]))
+            # 'copy' existing product_packaging for additional
+            # product_product records
+            for p in product_rows[1:]:
+                fields = values = column_names
+                cr.execute("""
+                    INSERT INTO product_packaging %s
+                    SELECT %s
+                    FROM product_packaging WHERE id = %s;
+                """, (tuple(fields.append('product_id')),
```
Comment (StefanRijnhart) [inline addons/product/migrations/11.0.1.2/post-migration.py]: Now 'product_id' is also added to fields and column_names. Try `fields + ['product_id']`

File: addons/product/migrations/11.0.1.2/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                      ','.join(values.append(p[0])), r[0]))
```
Comment (StefanRijnhart) [inline addons/product/migrations/11.0.1.2/post-migration.py]: The lists of fields and values are instantiated as a single element in the SQL.

>>> session.env.cr.mogrify("SELECT %s FROM res_partner", (','.join(['id', 'name']),))
"SELECT 'id,name' FROM res_partner"

You'll want something like

```
cr.execute(
     """
    INSERT INTO product_packaging %s
    SELECT %s
    FROM product_packaging WHERE id = %%s""" % (
        ','.join(fields + ['product_id']), ','.join(values + [p[0]])),
    (r[0],))
```

**Comment by StefanRijnhart** [comment]
@benniphx will you be able to have look at the requests (and rebase on the main branch, so that tests are run on this PR)? Otherwise, let me know.

**Comment by pedrobaeza** [comment]
@Benniphx are you going to finish this PR or should we take it over?

**Comment by Benniphx** [comment]
I will continue today

**Comment by StefanRijnhart** [comment]
Needs a rebase because of the new column length in the module status table, and a flake8 fix (see travis output), otherwise good!

**Review by StefanRijnhart — CHANGES_REQUESTED** [review]
See previous comment

**Comment by Benniphx** [comment]
I am working on it and I have to change a part in sql function as well
I am on it

File: odoo/openupgrade/doc/source/modules100-110.rst [hunk 3]
```diff
@@ -1,5 +1,5 @@
-Module coverage 10.0 -> 11.0
-============================
+Module coverage 9.0 -> 10.0
```
Comment (MiquelRForgeFlow) [inline odoo/openupgrade/doc/source/modules100-110.rst]: Why this change?

Comment (Benniphx) [reply odoo/openupgrade/doc/source/modules100-110.rst]: ah this was from the rebase i thought i changed it back

**Comment by MiquelRForgeFlow** [comment]
Please rebase and squash

File: addons/product/migrations/11.0.1.2/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            # 'copy' existing product_packaging for additional
+            # product_product records
+            for p in product_rows[1:]:
+                fields = values = column_names
+                cr.execute("""
+                   INSERT INTO product_packaging %s
+                   SELECT %s
+                   FROM product_packaging WHERE id = %%s""" % (
+                    ','.join(fields + ['product_id']),
+                    ','.join(values + ['product_id'])), (r[0],))
```
Comment (Benniphx) [inline addons/product/migrations/11.0.1.2/post-migration.py]: @StefanRijnhart  I think i should use something like [p[0]] instead of ['product_id'] in the  ','.join(values + ['product_id'])), (r[0],)) line

but @rruebner thinks this will not work. Maybe a hint? or idear

Comment (Benniphx) [reply addons/product/migrations/11.0.1.2/post-migration.py]: ok Travis said it is ok so i changed it to [p[0]]

**Comment by Benniphx** [comment]
It was rebased and squashed and is now working correctly

**Review by StefanRijnhart — APPROVED** [review]
